src/main/scala/hercules/actors/notifiers/NotifierActor.scala

```diff
@@ -1,14 +1,65 @@
 package hercules.actors.notifiers
 
-import akka.actor.Actor
-import akka.actor.ActorLogging
-import hercules.protocols.HerculesMainProtocol.HerculesMessage
-import hercules.entities.notification.NotificationUnit
+import akka.actor.{ ActorSystem, ActorRef, Actor, ActorLogging }
+import akka.event.LoggingReceive
+import hercules.config.notification.NotificationConfig
+import hercules.protocols.HerculesMainProtocol.{ SendNotificationUnitMessage, FailedNotificationUnitMessage }
+import scala.concurrent.duration._
 
 /**
  * Send notifications of events (e.g. email them or push cards around on a
  * trello board). All specific implementations of notifier Actors should extend
  * the NotifierActor trait. Note that it does not extend the HerculesActor since
- * that would cause circular dependencies (?)
+ * that would cause circular dependencies.
  */
-trait NotifierActor extends Actor with ActorLogging
+trait NotifierActor extends Actor with ActorLogging {
+
+  /**
+   * The executor which will be used to to the actual lifting of the notifier actor.
+   * @return a reference to a executor actor
+   */
+  def executor: ActorRef
+
+  /**
+   * The configuration for this particular notifier.
+   * @return a configuration for the notifier
+   */
+  def notifierConfig: NotificationConfig
+
+  /**
+   * Check if this particular notifier is subscribing to the channel
+   * on which the message is being sent, and if so forward it to the
+   * executor.
+   *
+   * Will also handle sending retries up until the specified maximum
+   * number of tries.
+   *
+   * Handles messages of types:
+   * SendNotificationUnitMessage, FailedNotificationUnitMessage
+   */
+  override def receive: Receive = LoggingReceive {
+
+    case failedNotificationUnit: FailedNotificationUnitMessage => {
+      import context.dispatcher
+
+      if (failedNotificationUnit.unit.attempts <= notifierConfig.numRetries)
+        context.system.scheduler.scheduleOnce(
+          notifierConfig.retryInterval.seconds,
```

Comment:
Just to note - I don't know if this is necessary, and if so how to do it. So I'll leave it for now.